spark/util/decorators.py

```diff
@@ -18,9 +21,13 @@ def wrapper(*args, **kwargs):
 
 def global_immut(func):
```

Comment:
Because global methods either belong to the immutable or mutable set it seems that a simpler choice would be to have a single mapping from method name to its corresponding value of immutable vs immutable. Then perhaps it makes sense to have a single decorator function which accepts an argument defining whether it should be mutable or immutable.

Reply:
Certainly reducing it to the single mapping makes sense, perhaps something like this?
```python
def ignite_global(*, mutable = False):
    def decorator(func):
        global ignite_globals
        if func.__name__ not in ignite_globals:
            ignite_globals[func.__name__] = mutable
        elif not mutable:
            raise RuntimeError("Attempted redefinition of a function marked as mutable")
        return func
    return decorator
```
With, of course, the appropriate changes made to ensure compatibility with the recent `@extern` decorator.

Reply:
Personally, while I agree with changing it to be a single mapping, I don't think it's strictly necessary to change it to use only one decorator. Having two decorators makes it very easy to see whether it's mutable or immutable at a glance, as opposed to remembering the convention of "immutable unless specified otherwise". That said, not a big deal either way, and I don't think there's a significant performance impact, as it should all be at import time.